### ISSUE TITLE
Set destination using search text if exact match is found

### ISSUE THREAD

**Issue by InappropriatePenguin** [EarendelDevelopers/factorio-mods#227]
If a player hits `<Enter>` while in a destination search field _and_ their search text is an exact match for a zone/spaceship name _and_ that zone/spaceship exists in the dropdown (ie. is a valid target for the relevant entity/force), then set the destination to the player's search text.

Modified from original suggestion by stringweasel:
> When searching a destination, if there is only one match left, automatically select it. Keep the cursor in the search though. It's quite annoying have to type in name, having only the one location left, but still having to click on the drop down to select it.

**Comment by Wiwiweb** [comment]
Does it have to be exact match? I like the suggestion of doing it if there's only 1 match in the dropdown.

**Comment by InappropriatePenguin** [comment]
~~The problem with that is that you could never select "Nauvis" because "Nauvis Orbit" would always be a possible match.~~

Actually, if you were talking about it in the context of `on_gui_confirmed`, it would be fine.

**Comment by Wiwiweb** [comment]
Good point. Yeah I meant it would apply if you press enter (on_gui_confirmed)
Maybe both then? Either exact match or there's only 1 non-exact match?
Could be useful in some cases like if a player wants to go to an orbit, or if they hid specific orbits from the list using the priority filter.

![image](https://user-images.githubusercontent.com/4723472/183217748-e7c27cb4-8fe6-49d7-8340-157ae778074d.png)
